### DISCUSSION TITLE
Brewing PyPy

### DISCUSSION THREAD

**Discussion by dwt** [Homebrew/discussions#720] [Everyday usage] [answered]
Hi there,

trying to get pypy building on Big Sur, I noticed that it is basically downloading a [pre-built version of pypy here](https://github.com/Homebrew/homebrew-core/blob/b024c23fcfcc26440a08dd0d325b289577d5617e/Formula/pypy.rb#L34) - just to then use it to compile that same binary itself.

I mean, it is nice that homebrew can compile the package given it already downloaded a compiled version of the package - but wouldn't it be much simpler to just install the downloaded package?

Or instead of depending on pypy itself to build itself, use the built in python2 interpreter from macOS and don't download anything?

**Comment by SMillerDev** [comment, marked as answer]
Some tools require a version of themselves to bootstrap (for example Java). Seeing how we still want to optimize the build for our use cases (file locations and compiler flags) we can't just use the original binary for that.

**Reply by dwt** [reply]
thanks

**Comment by carlocab** [comment]
Note that, last I checked, pypy isn't compatible with Big Sur yet: https://github.com/Homebrew/homebrew-core/pull/68019

That may have changed since, but, if that were the case, the formula would need a revision bump.

**Edit:** Nope, still an open issue: https://foss.heptapod.net/pypy/pypy/-/issues/3314 However, the last comment suggests that a `HEAD` build might work.

**Reply by dwt** [reply]
Also See: https://foss.heptapod.net/pypy/pypy/-/issues/3392#note_153618

Currently building from head works, if you hand-modify the pre-built version you download to actually build it.

So my understanding is: We need to wait for the next release of pypy to actually include the fixes, then we can modify the formula to download the current version (instead of the previous version) and use that to build pypy itself.

As I said above, seems a bit redundant just to customise the install location…

**Reply by SMillerDev** [reply]
We wouldn't be building the HEAD version, so even if we take a nightly binary it wouldn't build.
